### ISSUE TITLE
Enable Go linters

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/slsa-verifier#363]

**Comment by asraa** [comment]
deadcode is deprecated and replaced by unused. just FYI

**Comment by ianlewis** [comment]
> deadcode is deprecated and replaced by unused. just FYI

Yeah, we can just remove it and `varcheck` (though I see you removed it already) since we already have `unused` enabled.

**Comment by asraa** [comment]
> Yeah, we can just remove it and varcheck (though I see you removed it already) since we already have unused enabled.

Weirdly enough, it still seemed to be working. I removed some deadcode because of it's alert, but when I remove the deadcode check and introduce it again, it doesn't complain. It seems to be triggering either some form of unused w/ deadcode, OR it is still working but is staged to be deprecated. I'll investigate if I can remove it in a further PR

**Comment by ianlewis** [comment]
Removed deadcode here: https://github.com/slsa-framework/slsa-verifier/pull/572/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9L24
